src/QirRuntime/test/QIR-tracer/tracer-config.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// TODO: ideally, this file should be generated by the Q# compiler alongside the qir, using the mappings specified in
+// target.qs.
+
+#include <memory>
+
+#include "QuantumApi_I.hpp"
+#include "tracer-config.hpp"
+
+namespace TracerUser
+{
+const std::unordered_map<OpId, std::string> g_operationNames = {{0, "X"}, {1, "CX"}, {2, "MCX"},
+                                                                {3, "Y"}, {4, "CY"}, {5, "MCY"} /*etc.*/};
+}
+
+namespace Microsoft
+{
+namespace Quantum
+{
+
+    std::unique_ptr<ISimulator> CreateFullstateSimulator()
+    {
+        throw std::logic_error("Tracer should not instantiate full state simulator");
+    }
+
+} // namespace Quantum
+} // namespace Microsoft
```

Comment:
What will happen if this block is removed?
(we are creating an opportunity to make a mistake (we provide a function) and then we punish for making a mistake (for calling that function))

Reply:
It would fail to link.

This is an artefact of how QIR support is wired in if the clients consume the runtime using qdk.dll ad the qdk creates the default (fullstate) simulator for them. This is a sign of bad dependencies between the components and we should look into it, but not as part of this PR. Logged as https://github.com/microsoft/qsharp-runtime/issues/519.